chapters/modules/chapter_1/pages/environment_setup.adoc

```diff
@@ -1,7 +1,162 @@
 [id="environment"]
 
 [#setup]
-= Setting up your environment
+= Setting up your local development environment
+
+This section will cover three steps: 
+
+* Installing prerequisites
+* Installing the `cairo-lang` CLI
+* Installing the `cairo` Cairo 1 compiler
+
+== Pre-Requisites
+
+In order to install and use `cairo` and `cairo-lang`, install the following: +
+
+* https://www.python.org/downloads/release/python-390/[Python 3.9] +
+* https://www.rust-lang.org/tools/install[Rust] +
+
+== `cairo-lang` CLI Installation
+
+=== +++<strong>+++Step 1: Set up your virtual environment +++</strong>+++
+
+We recommend working inside a Python virtual environment, but you can also install the Cairo package directly. To create and enter the virtual environment, type:
+
+[,Bash]
+----
+#Activate your cairo virtual environment
+python3.9 -m venv ~/cairo_venv
+source ~/cairo_venv/bin/activate
+----
+
+Alternatively, you can also use the pyenv environment manager to create a virtual environment. To do so, run the following commands:
+
+[,Bash]
+----
+#Install pyenv
+curl https://pyenv.run | bash
+
+#Add the following to your .bashrc or .zshrc
+export PATH="$HOME/.pyenv/bin:$PATH"
+eval "$(pyenv init -)"
+eval "$(pyenv virtualenv-init -)"
+
+#Restart your terminal and run the following commands
+pyenv install 3.9.0
+pyenv virtualenv 3.9.0 cairo_venv
+pyenv activate cairo_venv
+----
+
+Either way, make sure the venv is activated – you should see (cairo_venv) in the command line prompt. 
+Your terminal should look like this:
+
+[,Bash]
+----
+(cairo_venv) satoshi@nakamoto-notebook % 
+image::cairo_venv.png[cairo venv]
+
+----
+
+=== +++<strong>+++Step 2: Install the necessary dependencies +++</strong>+++
+
+On Ubuntu, for example, you will have to first run:
+
+[,Bash]
+----
+sudo apt install -y libgmp3-dev
+----
+
+On Mac, you can use brew:
+
+[,Bash]
+----
+brew install gmp
+----
+
+Finally, install the following dependencies:
+
+[,Bash]
+----
+(cairo_venv) satoshi@nakamoto-notebook % pip install ecdsa fastecdsa sympy
```

Comment:
This step is not really needed. Installing `cairo-lang` will install all those dependencies automatically

Reply:
Are you sure? I faced some gmp issues while installing cairo-lang so, had to install it separately. Also, I have an M2, so I find many things are buggy :(
Maybe I will keep it as optional dependencies?